runtime-modules/argo-bridge/src/lib.rs

```diff
@@ -0,0 +1,221 @@
+// Ensure we're `no_std` when compiling for Wasm.
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(
+    not(any(test, feature = "runtime-benchmarks")),
+    deny(clippy::panic),
+    deny(clippy::panic_in_result_fn),
+    deny(clippy::unwrap_used),
+    deny(clippy::expect_used),
+    deny(clippy::indexing_slicing),
+    deny(clippy::integer_arithmetic),
+    deny(clippy::match_on_vec_items),
+    deny(clippy::unreachable)
+)]
+
+#[cfg(not(any(test, feature = "runtime-benchmarks")))]
+#[allow(unused_imports)]
+#[macro_use]
+extern crate common;
+
+use core::{convert::TryInto, default::Default};
+use frame_support::{
+    decl_module, decl_storage,
+    dispatch::{marker::Copy, DispatchResult},
+    ensure,
+    storage::bounded_vec::BoundedVec,
+    traits::{ConstU32, Currency, Get},
+};
+use frame_system::{ensure_root, ensure_signed};
+
+use sp_std::vec;
+
+// crate modules
+mod benchmarking;
+mod errors;
+mod events;
+mod tests;
+pub mod types;
+pub mod weights;
+pub use weights::WeightInfo;
+
+// crate imports
+use common::costs::{burn_from_usable, has_sufficient_balance_for_payment};
+pub use errors::Error;
+pub use events::{Event, RawEvent};
+use types::*;
+type WeightInfoArgo<T> = <T as Config>::WeightInfo;
+
+pub trait Config: frame_system::Config + balances::Config {
+    // /// Events
+    type RuntimeEvent: From<Event<Self>> + Into<<Self as frame_system::Config>::RuntimeEvent>;
+
+    type MaxPauserAccounts: Get<u32>;
+
+    /// Weight information for extrinsics in this pallet.
+    type WeightInfo: WeightInfo;
+
+    /// Defines the default bridging fee.
+    type DefaultBridgingFee: Get<BalanceOf<Self>>;
+}
+
+decl_storage! { generate_storage_info
+    trait Store for Module<T: Config> as ArgoBridge {
+        pub Status get(fn status) config(): BridgeStatus<T::BlockNumber>;
+
+        /// Account ID that operates the bridge
+        pub OperatorAccount get(fn operator_account): Option<T::AccountId>;
+
+        /// List of account IDs with permission to pause the bridge operations
+        pub PauserAccounts get(fn pauser_accounts): BoundedVec<T::AccountId, T::MaxPauserAccounts>;
+
+        /// Number of tokens that the bridge pallet is able to mint
+        pub MintAllowance get(fn mint_allowance) config(): BalanceOf<T>;
```

Comment:
I believe you can set default value of this to `0` if I correctly understood the bridging accounting logic, as there shouldn't be any other possible initial value

Reply:
in the tests is handy to be able to set a value for the mint allowance in the config

Reply:
or were you really just suggesting a default value and not removing it from the genesis config?

Reply:
the storage value mint_allowance is increased during Outbound transfer and decreased during Inbound transfer and it represent the total amount of JOY we have outside of the Joystream mainnet network.
If so my observation was: since the only plausible initial value is zero, we can set the default value when declearing the storage variable inside the decl_storage macro instead of manually setting the value in the chainspec config. (Even though I remember with confidence that in case a storage variable is not initialised it will be set to the fault value when accessed for the first time, so 0 for mint allowance which in production will be a u128)

Reply:
this is a minor comment however, I am fine even with the existing solution